python/ray/tune/trial_executor.py

```diff
@@ -194,6 +215,31 @@ def on_step_end(self, trials: List[Trial]) -> None:
     def force_reconcilation_on_next_step_end(self) -> None:
         pass
 
+    def may_warn_insufficient_resources(self, all_trials):
+        autoscaler_enabled = is_ray_cluster()
+        if not any(trial.status == Trial.RUNNING for trial in all_trials):
+            if self._no_running_trials_since == -1:
+                self._no_running_trials_since = time.monotonic()
+            elif time.monotonic(
+            ) - self._no_running_trials_since > _get_warn_threshold(
+                    autoscaler_enabled):
+                warn_prefix = ("If autoscaler is still scaling up, ignore "
+                               "this message." if autoscaler_enabled else
+                               "Autoscaler is disabled.")
+                logger.warning(
+                    f"{warn_prefix} "
+                    f"No trial is running and no new trial has been started "
+                    f"within at least the last "
+                    f"{_get_warn_threshold(autoscaler_enabled)} seconds. "
+                    f"This could be due to the cluster not having enough "
+                    f"resources available to start the next trial. Please "
+                    f"check if the requested resources can be fulfilled by "
+                    f"your cluster, or will be fulfilled eventually (when "
+                    f"using the Ray autoscaler).")
```

Comment:
A couple comments here:

1. IMO users that aren't using the Ray autoscaler should not see "Ray autoscaler"
2. This doesn't actually provide any action for the user to take. For example, user may not know what "requested resources" means nor even "cluster". 

Instead, it would be good to say:

1. which resource is not available, and how much is being requested
2. what is the total amount of those resource available on the cluster

Also, one suggestion should be to say that they should stop their tuning job and reconfigure their resource request.

Does that make sense? In principle, we should provide 1. what went wrong on the Ray side (in terms that the end user understands) 2. what the user did wrong (if possible) 3. what they should do instead :)

Reply:
This is good feedback. 
Practically there is no API that exposes those information. Left a TODO and filed https://github.com/ray-project/ray/issues/17799 to follow up.